### ISSUE TITLE
android_resource_apk not being passed from Android Gradle Plugin 3.2

### ISSUE THREAD

**Issue by alexblack** [robolectric/robolectric#4076]
I'm using Robolectric 4.0.1 with Android Studio 3.2 and latest gradle and plugin.  I'm writing my first unit test with Robolectric, and its complaining that I'm using legacy resources mode.

One of the modules i'm using is also complaining about missing permission that I actually have in our manifest.

I dug into the RobolectricTestRunner source, and it seems to think the manifest(s) don't support binary resource mode because `apkFile` is null, but I didn't trace back to see why that is the case.  Note: our project uses flavors.

```
java.lang.Exception: Failed: To use this functionality, add this to your AndroidManifest.xml:
<uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
```

```
import android.content.Context;
import org.junit.Test;
import org.junit.runner.RunWith;

import org.robolectric.RobolectricTestRunner;
import org.robolectric.annotation.Config;

import java.util.HashMap;
import java.util.Map;

import androidx.test.core.app.ApplicationProvider;
import bolts.Task;

@RunWith(RobolectricTestRunner.class)
@Config(application = UnitTestApp.class)
public class ParseInitTest  {

  @Test
  public void testQuery() throws Exception {
    final Context ctx = ApplicationProvider.getApplicationContext();

    final ParseInit parseInit = new ParseInit(ctx);

    final Map<String, String> businessInfo = new HashMap<>();
    final Task<Void> task = parseInit.createGuestUserAndAccount(businessInfo);
    task.waitForCompletion();

    if (task.isFaulted()) {
      throw new Exception("Failed: "+ task.getError().getMessage(), task.getError());
    }
  }
}

```

Our `build.gradle` has this indented within `android`:

```
    // http://google.github.io/android-gradle-dsl/current/com.android.build.gradle.internal.dsl.TestOptions.UnitTestOptions.html
    testOptions {
        unitTests {
            // returnDefaultValues = true
            includeAndroidResources = true // http://robolectric.org/migrating/#migrating-to-40
            all {
                jvmArgs '-noverify' // https://github.com/robolectric/robolectric-gradle-plugin/issues/144
                systemProperty 'http.agent', 'test-user-agent'
            }
        }
    }
```

Our `gradle.properties`:

```
# Project-wide Gradle settings.

# IDE (e.g. Android Studio) users:
# Gradle settings configured through the IDE *will override*
# any settings specified in this file.

# For more details on how to configure your build environment visit
# http://www.gradle.org/docs/current/userguide/build_environment.html

# Specifies the JVM arguments used for the daemon process.
# The setting is particularly useful for tweaking memory settings.
# Default value: -Xmx10248m -XX:MaxPermSize=256m
# org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8

# When configured, Gradle will run in incubating parallel mode.
# This option should only be used with decoupled projects. More details, visit
# http://www.gradle.org/docs/current/userguide/multi_project_builds.html#sec:decoupled_projects
# org.gradle.parallel=true

# http://stackoverflow.com/questions/23367769/how-to-speed-up-android-studio-compilation-process
org.gradle.daemon=true
org.gradle.jvmargs=-Xmx2048m -XX:MaxPermSize=512m -XX:+HeapDumpOnOutOfMemoryError -Dfile.encoding=UTF-8
org.gradle.parallel=true
org.gradle.configureondemand=true
android.useDeprecatedNdk=true
# http://robolectric.org/migrating/#migrating-to-40
android.enableUnitTestBinaryResources=true 
```

**Comment by jongerrish** [comment]
Can you put a breakpoint in DefaultManifestFactory.identify() and let us know the value of the property

android_resource_apk

Be interested to know if Gradle is populating this or if it is but the file doesn't exist

**Comment by jongerrish** [comment]
@alexblack - Right that's what I meant, and it _should_ have a value supplied by the AGP (you can see that AGP is providing us the values for legacy resources there).

Interestingly https://developer.android.com/studio/releases/ seems to suggest that you need to explicitly specify the build tools version to get another feature enabled and I wonder if this is related:-

"""
Although you typically don't need to specify the build tools version, when using Android Gradle plugin 3.2.0 with renderscriptSupportModeEnabled set to true, you need to include the following in each module's build.gradle file:

android.buildToolsVersion "28.0.3"
"""

Can you give that a try and if it still doesn't work do you mind creating a bug on Android Studio here:-

https://issuetracker.google.com/issues/new?component=192708&template=840533

..and link in this issue to that bug and vice versa?

Thanks

**Comment by alexblack** [comment]
Hmm. Ok in our android app (module)  `build.gradle` we didn't have `buildToolsVersion` specified, so I added it, re-tested, no change.

In our project `build.gradle`, we already have this code:

```
// Fix for projects like react-native-image-crop-picker and realm
// see https://github.com/oblador/react-native-vector-icons/issues/428
// https://medium.com/@alberto.schiabel/how-to-permanently-fix-sdk-build-tools-is-too-low-for-project-error-in-react-native-for-android-1a2d245bda18
subprojects { subproject ->
    afterEvaluate {
        if ((subproject.plugins.hasPlugin('android') || subproject.plugins.hasPlugin('android-library'))) {
            android {
                compileSdkVersion 26
                buildToolsVersion "28.0.3"
            }
        }
    }
}
```

This fixes up some of our 3rd party dependencies which specify older `buildToolsVersion`

**Comment by jongerrish** [comment]
@alexblack I'm not sure if it should fix the binary resources issue or not, just thought it was interesting that on the release notes for AGP it was called out as required for another configuration property. This _should_ work, but the problem is in Android Studio w/ Gradle.

Mind filing a bug on their issue tracker to report this?

**Comment by alexblack** [comment]
@jongerrish sure thing. Can you re-summarize for me what the issue is I should be filing? I'm a bit out of my depth here.

**Comment by jongerrish** [comment]
@alexblack were you able to open an issue on google issue tracker for studio?

**Comment by alexblack** [comment]
I opened an issue here: https://issuetracker.google.com/issues/119156712

**Comment by jongerrish** [comment]
Thank you 😃

**Comment by iRYO400** [comment]
How is it going? 
Tested on both studio 3.2.1 and 3.3 RC3 and get this 
`[Robolectric] NOTICE: legacy resources mode is deprecated; see http://robolectric.org/migrating/#migrating-to-40`
If I add `android.enableUnitTestBinaryResources=true` to gradle.properties Android Studio starts complaining
```
WARNING: The option setting 'android.enableUnitTestBinaryResources=true' is experimental and unsupported.
The current default is 'false'
Affected Modules: app
```

**Comment by Razzlegames** [comment]
Adding `android.enableUnitTestBinaryResources=true` in `gradle.properties` fixes this issue. But, this is experimented/not supported?  They closed the Google bug as "WontFix Obsolete". 

 Still occurs in:  
*Android Studio 3.3.2*
*Gradle 3.3.2*
`testImplementation 'org.robolectric:robolectric:4.2'`

Of course I have the flag set in `build.gradle` as indicated on RoboElectric docs:

```
 testOptions {
        unitTests {
            includeAndroidResources = true
        }
    }
```

*Is there any real fix to this? *

**Comment by AstralStorm** [comment]
The fix is to use plugin 3.4.1. Which is unfortunately incompatible with Android Studio older than 3.3.1.

**Comment by utzcoz** [comment]
Please try the latest AGP with latest Robolectric, this issue might be fixed. Feel free to reopen this issue if it still exists with latest AGP and Robolectric.
